exporter/mezmoexporter/exporter.go

```diff
@@ -0,0 +1,174 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mezmoexporter
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+	"sync"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+)
+
+type mezmoExporter struct {
+	config   *Config
+	settings component.TelemetrySettings
+	client   *http.Client
+	wg       sync.WaitGroup
+}
+
+type MezmoLogLine struct {
+	Timestamp int64             `json:"timestamp"`
+	Line      string            `json:"line"`
+	App       string            `json:"app"`
+	Level     string            `json:"level"`
+	Meta      map[string]string `json:"meta"`
+}
+
+type MezmoLogBody struct {
+	Lines []MezmoLogLine `json:"lines"`
+}
+
+func newLogsExporter(config *Config, settings component.TelemetrySettings) *mezmoExporter {
+	var e = &mezmoExporter{
+		config:   config,
+		settings: settings,
+	}
+	return e
+}
+
+func (m *mezmoExporter) pushLogData(ctx context.Context, ld plog.Logs) error {
+	m.wg.Add(1)
+	defer m.wg.Done()
+
+	return m.logDataToMezmo(ld)
+}
+
+func (m *mezmoExporter) start(_ context.Context, host component.Host) (err error) {
+	client, err := m.config.HTTPClientSettings.ToClient(host.GetExtensions(), m.settings)
+	if err != nil {
+		return err
+	}
+
+	m.client = client
+
+	return nil
+}
+
+func (m *mezmoExporter) stop(context.Context) (err error) {
+	m.wg.Wait()
+	return nil
+}
+
+func (m *mezmoExporter) logDataToMezmo(ld plog.Logs) error {
+	var errs error
+
+	var lines []MezmoLogLine
+
+	// Convert the log resources to mezmo lines...
+	resourceLogs := ld.ResourceLogs()
+	for i := 0; i < resourceLogs.Len(); i++ {
+		ills := resourceLogs.At(i).ScopeLogs()
+		//resource := resourceLogs.At(i).Resource()
+		for j := 0; j < ills.Len(); j++ {
+			logs := ills.At(j).LogRecords()
+
+			for k := 0; k < logs.Len(); k++ {
+				var log = logs.At(k)
+
+				// Convert Attributes to meta fields being mindful of the maxMetaDataSize restriction
+				var attrs = map[string]string{}
```

Comment:
```suggestion
				attrs := map[string]string{}
```
nit: use conventional golang patterns